bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -116,11 +116,35 @@ private boolean isTargetBranchAllowed() {
             var issues = new ArrayList<Issue>();
             issues.add(issue.get());
             issues.addAll(SolvesTracker.currentSolved(pr.repository().forge().currentUser(), comments));
+            getCsrIssue(issue.get()).ifPresent(issues::add);
             return issues;
         }
         return List.of();
     }
 
+    private Optional<Issue> getCsrIssue(Issue issue) {
+        var issueProject = issueProject();
+        if (issueProject != null) {
+            var jbsIssue = issueProject.issue(issue.shortId());
+            if (jbsIssue.isEmpty()) {
+                return Optional.empty();
+            }
+            org.openjdk.skara.issuetracker.Issue csr = null;
+            for (var link : jbsIssue.get().links()) {
+                var relationship = link.relationship();
+                if (relationship.isPresent() && relationship.get().equals("csr for")) {
+                    csr = link.issue().orElseThrow(
+                            () -> new IllegalStateException("Link with title 'csr for' does not contain issue")
```

Comment:
I don't think we should throw an exception here. If we do, then a bad state in JBS will completely prevent the PullRequestBot from updating the status of the affected PR. I would prefer logging a warning and returning Optional.empty().